cdap-common/src/main/java/co/cask/cdap/common/namespace/DefaultNamespacedLocationFactory.java

```diff
@@ -47,7 +57,20 @@ public Location get(Id.Namespace namespaceId) throws IOException {
 
   @Override
   public Location get(Id.Namespace namespaceId, @Nullable String subPath) throws IOException {
-    Location namespaceLocation = locationFactory.create(namespaceDir).append(namespaceId.getId());
+    String hdfsDirectory = null;
+    try {
+      hdfsDirectory = namespaceQueryAdmin.get(namespaceId).getConfig().getHdfsDirectory();
+    } catch (Exception e) {
+      Throwables.propagate(e);
```

Comment:
@chtyim : Is this okay ? The namespaceQueryAdmin.get throws Exception from its existing signature but the underlying implementation can either throw IOException or NamespaceNotFoundException. To throw the exact exception from here we will need to change the signature there. But this also adds the overhead that everyone using namespacedLocationFactory.get have to handle NamespaceNotFoundException and its at a lot of places in unit tests. So I prefered to wrap it runtime. Let me know.